### PR TITLE
Doc: sg section separator

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Prompted by https://github.com/matplotlib/matplotlib/issues/25016 this changes all of our SG section separators to `# %%` and documents that this is our preference.

### CODE REVIEW

File: doc/devel/documenting_mpl.rst [hunk 1]
```diff
@@ -814,7 +814,9 @@ from having an plot generated insert "sgskip" somewhere in the filename.
 
 The format of these files is relatively straightforward.  Properly
 formatted comment blocks are treated as ReST_ text, the code is
-displayed, and figures are put into the built page.
+displayed, and figures are put into the built page.  Matplotlib uses the
+``# %%`` section separator so that IDEs will identify "code cells" to make
+it easy to re-run sub-sections of the example.
```
Comment (rcomer) [inline doc/devel/documenting_mpl.rst:819]: There is an example of a tutorial below this which still has the long line.

Comment (tacaswell) [reply doc/devel/documenting_mpl.rst:819]: 🤦🏻 I looked for our documentation on how to split sections and was a bit surprised we did not have any.  I am glad the problem was me and not our docs!  Thank you for catching this.

**Review by rcomer — APPROVED** [review]

**Review by jklymak — CHANGES_REQUESTED** [review]
I think this is fine, but I don't think we should merge until we discuss and make sure everyone is onboard with the change.  So I'm just blocking until a few more people weight in with up or down thumbs or checkmarks.  

My only objection is that I find it less readable, but it's not terrible.

**Review by timhoffm — APPROVED** [review]

**Review by kolibril13 — APPROVED** [review]

**Comment by QuLogic** [comment]
Going to ignore the Azure pipeline, as it timed out there, but didn't post the result ot GitHub, so I can't even restart it. Plus this is all docs.

**Pull request merged**